compressed_image_transport/src/compressed_publisher.cpp

```diff
@@ -263,14 +302,83 @@ void CompressedPublisher::publish(
 
         // Publish message
         publish_fn(compressed);
+      } else {
+        RCUTILS_LOG_ERROR(
+          "Compressed Image Transport - PNG compression requires 8/16-bit encoded color format (input format is: %s)", message.encoding.c_str());
+      }
+      break;
+    }
+    // TIFF Compression
+    case TIFF:
+    {
+      // Update ros message format header
+      compressed.format += "; tiff compressed ";
+      int res_unit = -1;
+      // See https://gitlab.com/libtiff/libtiff/-/blob/v4.3.0/libtiff/tiff.h#L282-284
+      if (config_.tiff_res_unit == "inch") {
+        res_unit = 2;
+      } else if (config_.tiff_res_unit == "centimeter") {
+        res_unit = 3;
+      } else if (config_.tiff_res_unit == "none") {
+        res_unit = 1;
+      } else {
+        RCLCPP_WARN(
+          logger_,
+          "tiff.res_unit parameter should be either 'inch', 'centimeter' or 'none'; "
+          "defaulting to 'inch'. Found '%s'", config_.tiff_res_unit.c_str());
+      }
+      params.reserve(3);
+      params.emplace_back(cv::IMWRITE_TIFF_XDPI);
+      params.emplace_back(config_.tiff_xdpi);
+      params.emplace_back(cv::IMWRITE_TIFF_YDPI);
+      params.emplace_back(config_.tiff_ydpi);
+      params.emplace_back(cv::IMWRITE_TIFF_RESUNIT);
+      params.emplace_back(res_unit);
+
+      // Check input format
+      if ((bitDepth == 8) || (bitDepth == 16) || (bitDepth == 32))
+      {
+
+        // OpenCV-ros bridge
+        try
+        {
+          cv_bridge::CvImageConstPtr cv_ptr = cv_bridge::toCvShare(message, nullptr, "");
+
+          // Compress image
+          if (cv::imencode(".tiff", cv_ptr->image, compressed.data, params))
+          {
+
+            float cRatio = (float)(cv_ptr->image.rows * cv_ptr->image.cols * cv_ptr->image.elemSize())
+                / (float)compressed.data.size();
```

Comment:
Prefer `static_cast<float>`

Reply:
This was copy pasted from what we do in jpeg/png, but I will correct it here

```suggestion
            float cRatio = static_cast<float>((cv_ptr->image.rows * cv_ptr->image.cols * cv_ptr->image.elemSize()))
                / static_cast<float>((float)compressed.data.size());
```